### PR TITLE
Add motivation and example to delayed docs

### PR DESCRIPTION
**Pull request by mrocklin**
This starts off the dask.delayed docs with a user-facing motivation and
example.

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @davclark , mostly in an attempt to sucker him in to reviewing dask documentation.  A task at which I think he would be quite valuable.

File: docs/source/delayed-overview.rst [hunk 1]
```diff
@@ -1,6 +1,86 @@
...
+to turn into tasks.  None of the ``inc``, ``double``, ``add`` or ``sum`` calls
+have happened yet, instead the object ``total`` is a ``Delayed`` result that
+contains a task graph of the entire computation.  Looking at the graph we see
+clear opportunities for parallel execution.  The dask schedulers will exploit
+this parallelism, generally improving performance.  (although not in this
+example, because these functions are already very small and fast.)
+
+.. code-block:: python
+
+   total.visualize()  # see image to the right
```
Comment (davclark) [inline docs/source/delayed-overview.rst]: The image is below, actually, even at ludicrous browser widths (full screen on a 4K monitor):

http://dask.pydata.org/en/latest/delayed-overview.html

File: docs/source/delayed-overview.rst [hunk 2]
```diff
@@ -1,6 +1,86 @@
...
+
+As written this code runs sequentially in a single thread.  However we see that
+a lot of this could be executed in parallel.  We use the ``delayed`` function
+to parallelize this code by turning it into a dask graph.  We slightly modify
+our code by wrapping functions in ``delayed``.  This delays the execution of
+the function and generates a dask graph instead.
+
+.. code-block:: python
+
+   from dask import delayed
```
Comment (davclark) [inline docs/source/delayed-overview.rst]: For this example, I would find it easier to add the `@delayed ` decorator to the functions. Faster to type also... If folks don't understand decorators, they may also get confused by the syntax you use here. The decorator seems easier to use even if it just seems like magic.

File: docs/source/delayed-overview.rst [hunk 3]
```diff
@@ -1,6 +1,86 @@
 Overview
 ========
 
+Motivation and Example
+----------------------
+
+Dask.delayed lets you parallelize custom code.  It is useful whenever your
```
Comment (davclark) [inline docs/source/delayed-overview.rst]: Should Dask be capitalized here?

**Pull request merged**